lib/rmt/scc.rb

```diff
@@ -84,7 +84,8 @@ def sync_systems
     scc_api_client = SUSE::Connect::Api.new(Settings.scc.username, Settings.scc.password)
 
     # do not sync BYOS proxy systems to SCC
-    systems = System.where('scc_registered_at IS NULL OR last_seen_at > scc_registered_at').where(proxy_byos: false)
+    # using where in because `where not` does NOT work for enum on this Rails version
+    systems = System.where('scc_registered_at IS NULL OR last_seen_at > scc_registered_at').where(proxy_byos_mode: [nil, :hybrid])
```

Comment:
Why are we selecting `proxy_byos_mode: nil` ? `nil` is not defined in the _enum_, so therefore should not be considered a valid condition.

Reply:
Still querying on `proxy_byos_mode == nil` .

Reply:
I do not understand, I haven't committed any change yet, what do you mean with "still querying nil" ?

Reply:
the nil query must remain unless we use `nil, :payg, :byos, :hybrid`
Rails has the negative scope for enum borked
```bash
# emtpy DB
irb(main):022:0> System.all
D, [2024-08-01T10:54:39.143333 #9120] DEBUG -- :   System Load (0.7ms)  SELECT `systems`.* FROM `systems` /* loading for inspect */ LIMIT 11
=> #<ActiveRecord::Relation []>
irb(main):023:0> System.create(hostname: 'hostname', proxy_byos_mode: :byos)
D, [2024-08-01T10:54:57.262121 #9120] DEBUG -- :   TRANSACTION (0.3ms)  BEGIN
D, [2024-08-01T10:54:57.263324 #9120] DEBUG -- :   System Create (0.8ms)  INSERT INTO `systems` (`hostname`, `created_at`, `updated_at`, `proxy_byos_mode`) VALUES ('hostname', '2024-08-01 09:54:57', '2024-08-01 09:54:57', 1)
D, [2024-08-01T10:54:57.338011 #9120] DEBUG -- :   TRANSACTION (73.9ms)  COMMIT
=> #<System id: 5, login: nil, password: nil, hostname: "hostname", registered_at: nil, last_seen_at: nil, created_at: "2024-08-01 09:54:57.000000000 +0000", updated_at: "2024-08-01 09:54:57.000000000 +0000", scc_registered_at: nil, scc_system_id: nil, proxy_byos: false, system_token: nil, system_information: nil, instance_data: nil, proxy_byos_mode: "byos">
irb(main):024:0> System.where.not(proxy_byos_mode: :byos)
D, [2024-08-01T10:55:10.947890 #9120] DEBUG -- :   System Load (0.8ms)  SELECT `systems`.* FROM `systems` WHERE `systems`.`proxy_byos_mode` != 1 /* loading for inspect */ LIMIT 11
=> #<ActiveRecord::Relation []>
irb(main):026:0> System.create(hostname: 'hostname')
D, [2024-08-01T10:55:32.926403 #9120] DEBUG -- :   TRANSACTION (0.4ms)  BEGIN
D, [2024-08-01T10:55:32.927511 #9120] DEBUG -- :   System Create (0.7ms)  INSERT INTO `systems` (`hostname`, `created_at`, `updated_at`) VALUES ('hostname', '2024-08-01 09:55:32', '2024-08-01 09:55:32')
D, [2024-08-01T10:55:32.973673 #9120] DEBUG -- :   TRANSACTION (45.2ms)  COMMIT
=> #<System id: 6, login: nil, password: nil, hostname: "hostname", registered_at: nil, last_seen_at: nil, created_at: "2024-08-01 09:55:32.000000000 +0000", updated_at: "2024-08-01 09:55:32.000000000 +0000", scc_registered_at: nil, scc_system_id: nil, proxy_byos: false, system_token: nil, system_information: nil, instance_data: nil, proxy_byos_mode: nil>


# this should show the system with nil on proxy_byos_mode
irb(main):027:0> System.where.not(proxy_byos_mode: :byos)
D, [2024-08-01T10:55:36.176109 #9120] DEBUG -- :   System Load (0.8ms)  SELECT `systems`.* FROM `systems` WHERE `systems`.`proxy_byos_mode` != 1 /* loading for inspect */ LIMIT 11
=> #<ActiveRecord::Relation []>
irb(main):028:0> System.where(proxy_byos_mode: [:hybrid, nil])
D, [2024-08-01T10:55:38.069228 #9120] DEBUG -- :   System Load (0.9ms)  SELECT `systems`.* FROM `systems` WHERE (`systems`.`proxy_byos_mode` = 2 OR `systems`.`proxy_byos_mode` IS NULL) /* loading for inspect */ LIMIT 11
=> #<ActiveRecord::Relation [#<System id: 6, login: nil, password: nil, hostname: "hostname", registered_at: nil, last_seen_at: nil, created_at: "2024-08-01 09:55:32.000000000 +0000", updated_at: "2024-08-01 09:55:32.000000000 +0000", scc_registered_at: nil, scc_system_id: nil, proxy_byos: false, system_token: nil, system_information: nil, instance_data: nil, proxy_byos_mode: nil>]>
irb(main):029:0> 
```